### ISSUE TITLE
Add Principal informations in log.

### ISSUE THREAD

**Issue by dcdh** [quarkusio/quarkus#11927]
**Description**

I propose to enrich logs by adding the user who has made the action. So when a log is produced, we can ensure on who has made the action and trace hm easily. It is very useful for support team.

![image](https://user-images.githubusercontent.com/5189615/92335678-731a7e00-f099-11ea-9e25-104f468e8b09.png)

Regarding this screenshot, you can notice that it is already the case when tracing is activated (`spanId`, `traceId` are presents when the context support it and got it).

Maybe that this feature should be toggled on or off by adding a property in application.yml.

**Implementation ideas**

Data sent to the centralized log management are stored into the MDC.

I propose to follow what was done in tracing feature here https://github.com/quarkusio/quarkus/blob/master/extensions/jaeger/runtime/src/main/java/io/quarkus/jaeger/runtime/MDCScope.java

So maybe that we should add this kind of code in `quarkus-keycloak-authorization`:

```java
@Provider
public class ConnectedUserMDC implements ContainerRequestFilter {

    public static final String USER_ANONYMOUS = "user.anonymous";
    public static final String USER_NAME = "user.name";

    @Inject
    SecurityIdentity securityIdentity;

    @Override
    public void filter(final ContainerRequestContext requestContext) throws IOException {
        MDC.put(USER_ANONYMOUS, securityIdentity.isAnonymous());
        MDC.put(USER_NAME, Optional.ofNullable(securityIdentity)
                .map(SecurityIdentity::getPrincipal)
                .map(Principal::getName)
                .orElse(""));
    }

}
```

Regards,

Damien

**Comment by sberyozkin** [comment]
@dcdh I doubt this kind of enhancement will be supported, anything related to logging the security related information such as the currently logged in user's name will be considered as a potential security risk, or a data privacy risk. In a way, this risk can be somewhat of the academic nature in some specific closed/isolated deployments, but in most cases what we can expect is someone opening a security flaw/CVE request against Quarkus if we start shipping such a code.
I'll wait though for more comments, CC @pedroigor @stuartwdouglas, before closing this issue.
I think one can provide a custom filter, or security identity augmentor, and do it if it is needed

**Comment by stuartwdouglas** [comment]
Logging the current user is absolutely not a security issue, and is required by a lot of organisations for audit purposes. Logging credentials is a different matter but the username is absolutely fine.

This sort of code belongs in the jaeger extension though, not in Keycloak, and should not be tied to RESTEasy.

**Comment by sberyozkin** [comment]
Hi @stuartwdouglas I'm not sure; if it is `JWT` and Quarkus is configured to use a 'preferred_user_name' then we can get not just a login name like `alice1` but `Alice Brown` etc. If a given organization needs it then a filter like the one suggested above can be registered in no time; Quarkus doing it itself is a different story IMHO

**Comment by pedroigor** [comment]
I agree with @stuartwdouglas but I also see your point @sberyozkin, mainly in regards to privacy. Considering all the privacy regulations I'm not whether or not services should be able to track user activity without consent. I would then agree that for now, users should implement something like that filter. I'm afraid some one would open another issue later telling us to include a `quarkus.security.track-user-activity` flag :)

**Comment by sberyozkin** [comment]
Hi All, I'd like to close this issue but I'll work on the documentation update suggesting registering a custom filter as shown above if tracing the principal is needed.

**Comment by stuartwdouglas** [comment]
There are no privacy issues here, if you are using a system you should absolutely expect that there is a record of the actions you take in your system. This is a fundamental requirement of almost every business system.

**Comment by sberyozkin** [comment]
@stuartwdouglas I'm not going to close this issue.
As I have said above, it can leak a complete user name when JWT is used, not just `myloginname`. IMHO it is not something Quarkus should be doing, where it is acceptable the users can do it themselves

**Comment by sberyozkin** [comment]
Or maybe, if it is supported (since as Stuart @stuartwdouglas mentioned an access log %u is there to log it anyway) then we can just doc some warning that it should be done with care, something like that. So please don't mind me :-), if someone will open a PR (opentracing, etc) then I'll be happy to review as well (suggest the doc updates, etc)

**Comment by famod** [comment]
FWIW, we are implementing a German health related application and although it is not used for emergency cases, we have very strict compliance rules. One of these rules is that we are not allowed to log the username (loginname), but only the user's database id.

**Comment by rgmz** [comment]
> Or maybe, if it is supported (since as Stuart [@stuartwdouglas](https://github.com/stuartwdouglas) mentioned an access log %u is there to log it anyway) then we can just doc some warning that it should be done with care, something like that.

It appears that `%u` is no longer a valid format character[^1]. Is it safe to say we're back to the original suggestion of creating a manual filter?

[^1]: Not documented in [Logging Format](https://quarkus.io/guides/logging#logging-format) and throws an exception if used
